File: backend/src/data/ensureDemoData.ts
import type { PrismaClient } from '@prisma/client'
import { demoClubs } from './demoClubs'

export async function ensureDemoData(prisma: PrismaClient, enabled: boolean): Promise<number> {
  if (!enabled || await prisma.club.count() > 0) return 0
  await prisma.$transaction(demoClubs.map(data => prisma.club.create({ data })))
  return demoClubs.length
}

File: backend/app/data/ensure_demo_data.py
"""按需注入演示社团。等价于原 TS data/ensureDemoData.ts。"""
from __future__ import annotations

from sqlalchemy import func, select

from ..db import Club, SessionLocal
from .demo_clubs import DEMO_CLUBS


def ensure_demo_data(enabled: bool) -> int:
    if not enabled:
        return 0
    with SessionLocal() as session:
        count = session.scalar(select(func.count()).select_from(Club)) or 0
        if count > 0:
            return 0
        session.add_all([Club(**data) for data in DEMO_CLUBS])
        session.commit()
    return len(DEMO_CLUBS)
